sdk/iot/hub/src/az_iot_hub_client.c

```diff
@@ -172,3 +172,30 @@ AZ_NODISCARD az_result az_iot_hub_client_properties_find(
 
   return AZ_ERROR_ITEM_NOT_FOUND;
 }
+
+AZ_NODISCARD az_result
+az_iot_hub_client_properties_next(az_iot_hub_client_properties* properties, az_pair* out)
+{
+  AZ_PRECONDITION_NOT_NULL(properties);
+  AZ_PRECONDITION_NOT_NULL(out);
+
+  int32_t index = (int32_t)properties->_internal.current_property_index;
+  int32_t prop_length = az_span_length(properties->_internal.properties);
+
+  if (index == prop_length)
+  {
+    *out = AZ_PAIR_NULL;
+    return AZ_ERROR_EOF;
+  }
+
+  az_span remainder;
+  az_span prop_span = az_span_slice(properties->_internal.properties, index, prop_length);
+
+  out->key = az_span_token(prop_span, hub_client_param_equals_span, &remainder);
+  out->value = az_span_token(remainder, hub_client_param_separator_span, &remainder);
+  properties->_internal.current_property_index = az_span_length(remainder) == 0
```

Comment:
Please write the full if statement to make it easier to read and debug: your conditional is too complex for the short-hand notation.

Reply:
gotcha updated